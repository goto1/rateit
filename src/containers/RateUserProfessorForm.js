import React from "react";
import PropTypes from "prop-types";
import UserForm from "./UserForm";
import RatingCategoryInput from "../components/RatingCategoryInput";
import { FormSection, StyledContentBlock } from "./RateUser";
import Yup from "yup";
import {
  Button,
  ContentBlock,
  InputElement,
  SmartSelect
} from "../components/f7";

class ProfessorForm extends React.Component {
  renderRatingCategoriesList = () => {
    const { ratingCategories, handleChange, handleBlur } = this.props;

    return ratingCategories.map(cat => (
      <RatingCategoryInput
        key={cat.id}
        onChange={handleChange}
        onBlur={handleBlur}
        {...cat}
      />
    ));
  };

  render() {
    const props = this.props;
    const { values, errors, touched } = this.props;
    const ratingCategoriesList = this.renderRatingCategoriesList();

    return (
      <form onSubmit={props.handleSubmit}>
        <FormSection title="Professor Information">
          <InputElement
            icon="account_circle"
            name="name"
            onBlur={props.handleBlur}
            onChange={props.handleChange}
            placeholder="John Doe"
            type="text"
            valid={!errors.name && touched.name && true}
            value={values.name}
          />
        </FormSection>

        <FormSection title="School Information">
          <SmartSelect
            multiple={false}
            name="school"
            onBlur={props.handleBlur}
            onChange={e => props.handleSelectChange(e, props.handleChangeValue)}
            options={props.schools}
            searchbarPlaceholder="Search for a school..."
            value={values.school}
          />
          <SmartSelect
            multiple={false}
            name="major"
            onBlur={props.handleBlur}
            onChange={e => props.handleSelectChange(e, props.handleChangeValue)}
            options={props.majors}
            searchbarPlaceholder="Search for a major..."
            value={values.major}
          />
        </FormSection>

        {ratingCategoriesList}

        {!props.validForm && (
          <StyledContentBlock>Please fill out all fields</StyledContentBlock>
        )}

        <ContentBlock>
          <Button
            big
            color="green"
            disabled={props.disableSubmission}
            fill
            type="submit"
          >
            Submit
          </Button>
        </ContentBlock>
      </form>
    );
  }
}

const formikOptions = {
  mapPropsToValues: props => ({
    major: props.major,
    name: "",
    school: props.school,
    T5wKYAmI: "",
    jUtauYzO: "",
    mh4m4LcX: "",
    yZyycMRm: "",
    sBuPhZef: ""
  }),
  validationSchema: Yup.object().shape({
    major: Yup.string().required(),
    name: Yup.string().required(),
    school: Yup.string().required(),
    T5wKYAmI: Yup.string().required(),
    jUtauYzO: Yup.string().required(),
    mh4m4LcX: Yup.string().required(),
    yZyycMRm: Yup.string().required(),
    sBuPhZef: Yup.string().required()
  })
};

const RateUserProfessorForm = UserForm(ProfessorForm, formikOptions);

export default RateUserProfessorForm;
