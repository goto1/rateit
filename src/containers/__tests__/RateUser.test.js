import React from "react";
import RateUserProfessorForm from "../RateUserProfessorForm";
import RateUserStudentForm from "../RateUserStudentForm";
import {
  Button,
  InputElement,
  RadioInput,
  SmartSelect,
  Textarea
} from "../../components/f7";
import RatingCategoryInput from "../../components/RatingCategoryInput";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("RateUser container", () => {
  const getForm = type => {
    const props = {
      ratingCategories: [
        { id: "DRFJY9jd", description: "Attends Group Meetings" },
        { id: "j84WAR77", description: "Contributes to Discussions" }
      ],
      selectedSchools: [
        {
          id: "bruHy",
          name: "New Jersey Institute of Technology, Newark",
          abbreviation: "NJIT"
        }
      ],
      selectedMajors: [
        { id: "hvdMFjM", name: "Computer Science", abbreviation: "CS" }
      ],
      schools: [
        {
          id: "bruHy",
          name: "New Jersey Institute of Technology, Newark",
          abbreviation: "NJIT"
        },
        {
          id: "5y7fh",
          name: "Rutgers University, New Brunswick",
          abbreviation: "RU"
        }
      ],
      majors: [
        { id: "hvdMFjM", name: "Computer Science", abbreviation: "CS" },
        { id: "e2LejDg", name: "Information Systems", abbreviation: "IS" }
      ]
    };
    return type === "professor"
      ? <RateUserProfessorForm {...props} />
      : <RateUserStudentForm {...props} />;
  };

  describe("RateUserProfessorForm", () => {
    it("should render correctly", () => {
      const RateUserProfessorForm = getForm("professor");
      const tree = renderer.create(RateUserProfessorForm).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render an `InputElement` with the `name` attribute set to `name`", () => {
      const RateUserProfessorForm = getForm("professor");
      const wrapper = mount(RateUserProfessorForm);
      const actual = wrapper.find(InputElement).props().name;
      const expected = "name";
      expect(actual).toBe(expected);
    });

    it("should render two `SmartSelect` components with options for school and major information", () => {
      const RateUserProfessorForm = getForm("professor");
      const wrapper = mount(RateUserProfessorForm);
      const schools = wrapper.find(SmartSelect).at(0).props().name === "school";
      const majors = wrapper.find(SmartSelect).at(1).props().name === "major";
      const actual = schools && majors;
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render two `RatingCategoryInput` components", () => {
      const RateUserProfessorForm = getForm("professor");
      const wrapper = mount(RateUserProfessorForm);
      const actual = wrapper.find(RatingCategoryInput).length;
      const expected = 2;
      expect(actual).toBe(expected);
    });
  });

  describe("StudentForm", () => {
    it("should render correctly", () => {
      const RateUserStudentForm = getForm("student");
      const tree = renderer.create(RateUserStudentForm).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("should render two `InputElement` components", () => {
      const RateUserStudentForm = getForm("student");
      const wrapper = mount(RateUserStudentForm);
      const actual = wrapper.find(InputElement).length;
      const expected = 2;
      expect(actual).toBe(expected);
    });

    it("should render two `SmartSelect` components for picking school and major options", () => {
      const RateUserStudentForm = getForm("student");
      const wrapper = mount(RateUserStudentForm);
      const schools = wrapper.find(SmartSelect).at(0).props().name === "school";
      const majors = wrapper.find(SmartSelect).at(1).props().name === "major";
      const actual = schools && majors;
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render two `RatingCategoryInput` components", () => {
      const RateUserStudentForm = getForm("student");
      const wrapper = mount(RateUserStudentForm);
      const actual = wrapper.find(RatingCategoryInput).length;
      const expected = 2;
      expect(actual).toBe(expected);
    });

    it("should render a `Textarea` component with the `name` attribute of `comment`", () => {
      const RateUserStudentForm = getForm("student");
      const wrapper = mount(RateUserStudentForm);
      const actual = wrapper.find(Textarea).exists();
      const expected = true;
      expect(actual).toBe(expected);
    });

    it("should render a `RadioInput` component with the `name` attribute of `recommend`", () => {
      const RateUserStudentForm = getForm("student");
      const wrapper = mount(RateUserStudentForm);
      const actual = wrapper.find(RadioInput).props().name;
      const expected = "recommend";
      expect(actual).toBe(expected);
    });
  });
});
