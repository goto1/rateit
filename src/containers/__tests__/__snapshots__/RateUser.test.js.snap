// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateUser container ProfessorForm should render correctly 1`] = `
.jwltLt {
  background-color: #369947 !important;
  border-color: #369947 !important;
  color: #F1F1F5 !important;
  opacity: .65;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  width: 100%;
}

.cYrpQX {
  margin-top: 25px !important;
}

.fYkctb {
  margin-left: 15px;
}

.fYkctb input {
  color: black;
}

.bqyzQp {
  position: relative;
}

.eYQGSB {
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  margin: 0;
  cursor: pointer;
}

.djmajI {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border: 2px solid black;
  border-radius: 50%;
}

.djmajI label {
  font-weigth: 600;
  font-size: 17.5px;
  border-color: black;
}

.fbgGmL {
  position: relative;
  height: 25px;
  width: 25px;
  margin: 5px;
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
  -webkit-transform: ;
  -ms-transform: ;
  transform: ;
}

.cNVoic {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.dAvTTg .header {
  -webkit-text-align: center;
  text-align: center;
  font-weight: 500;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  margin-bottom: 5px;
}

.dAvTTg .footer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  margin-top: 7.5px;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 11.5px;
  color: #646464;
}

.dAvTTg .footer > div:nth-of-type(1) > .icon {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.enMCSC {
  margin-bottom: 25px !important;
}

<form
  onSubmit={[Function]}
>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      Professor Information
    </div>
    <div
      className="list-block inset sc-kAzzGY enMCSC"
    >
      <ul>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              account_circle
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="John Doe"
              type="text"
              value=""
            />
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      School Information
    </div>
    <div
      className="list-block inset sc-kAzzGY enMCSC"
    >
      <ul>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a school..."
          >
            <select
              multiple={false}
              name="school"
              onBlur={[Function]}
              onChange={[Function]}
              value="[object Object]"
            >
              <option
                data-display-as="NJIT"
                value="bruHy"
              >
                New Jersey Institute of Technology, Newark
              </option>
              <option
                data-display-as="RU"
                value="5y7fh"
              >
                Rutgers University, New Brunswick
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  School
                </div>
                <div
                  className="item-after"
                />
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a major..."
          >
            <select
              multiple={false}
              name="major"
              onBlur={[Function]}
              onChange={[Function]}
              value="[object Object]"
            >
              <option
                data-display-as="CS"
                value="hvdMFjM"
              >
                Computer Science
              </option>
              <option
                data-display-as="IS"
                value="e2LejDg"
              >
                Information Systems
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  Major
                </div>
                <div
                  className="item-after"
                />
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-content sc-jzJRlG dAvTTg"
    >
      <div
        className="card-content-inner"
      >
        <div
          className="header"
        >
          Attends Group Meetings
        </div>
        <div
          className="sc-fjdhpX cNVoic"
        >
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#D91518"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="1_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#D91518"
            >
              <label
                htmlFor="1_DRFJY9jd"
              >
                1
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#E6665B"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="2_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#E6665B"
            >
              <label
                htmlFor="2_DRFJY9jd"
              >
                2
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#DEB21C"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="3_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#DEB21C"
            >
              <label
                htmlFor="3_DRFJY9jd"
              >
                3
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#BCC75E"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="4_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#BCC75E"
            >
              <label
                htmlFor="4_DRFJY9jd"
              >
                4
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#A1BA37"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="5_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#A1BA37"
            >
              <label
                htmlFor="5_DRFJY9jd"
              >
                5
              </label>
            </div>
          </div>
        </div>
        <div
          className="footer"
        >
          <div>
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
             Awful
          </div>
          <div>
            Great 
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-content sc-jzJRlG dAvTTg"
    >
      <div
        className="card-content-inner"
      >
        <div
          className="header"
        >
          Contributes to Discussions
        </div>
        <div
          className="sc-fjdhpX cNVoic"
        >
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#D91518"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="1_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#D91518"
            >
              <label
                htmlFor="1_j84WAR77"
              >
                1
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#E6665B"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="2_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#E6665B"
            >
              <label
                htmlFor="2_j84WAR77"
              >
                2
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#DEB21C"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="3_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#DEB21C"
            >
              <label
                htmlFor="3_j84WAR77"
              >
                3
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#BCC75E"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="4_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#BCC75E"
            >
              <label
                htmlFor="4_j84WAR77"
              >
                4
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#A1BA37"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="5_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#A1BA37"
            >
              <label
                htmlFor="5_j84WAR77"
              >
                5
              </label>
            </div>
          </div>
        </div>
        <div
          className="footer"
        >
          <div>
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
             Awful
          </div>
          <div>
            Great 
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="content-block  "
  >
    <button
      className="button button-big sc-bdVaJa jwltLt"
      disabled={true}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`RateUser container StudentForm should render correctly 1`] = `
.dYzSUq {
  background-color: #369947 !important;
  border-color: #369947 !important;
  color: #F1F1F5 !important;
  opacity: 1;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  width: 100%;
}

.cYrpQX {
  margin-top: 25px !important;
}

.fYkctb {
  margin-left: 15px;
}

.fYkctb input {
  color: black;
}

.bqyzQp {
  position: relative;
}

.kmibPL {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
}

.kaKmBo {
  -webkit-text-align: right;
  text-align: right;
  width: auto !important;
}

.irilpd {
  position: relative;
}

.eYQGSB {
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  margin: 0;
  cursor: pointer;
}

.djmajI {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border: 2px solid black;
  border-radius: 50%;
}

.djmajI label {
  font-weigth: 600;
  font-size: 17.5px;
  border-color: black;
}

.fbgGmL {
  position: relative;
  height: 25px;
  width: 25px;
  margin: 5px;
  -webkit-transition: -webkit-transform 300ms ease;
  -webkit-transition: transform 300ms ease;
  transition: transform 300ms ease;
  -webkit-transform: ;
  -ms-transform: ;
  transform: ;
}

.cNVoic {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.dAvTTg .header {
  -webkit-text-align: center;
  text-align: center;
  font-weight: 500;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  margin-bottom: 5px;
}

.dAvTTg .footer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  margin-top: 7.5px;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 11.5px;
  color: #646464;
}

.dAvTTg .footer > div:nth-of-type(1) > .icon {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.enMCSC {
  margin-bottom: 25px !important;
}

.kqlxdJ {
  margin-bottom: 5px !important;
}

<form
  onSubmit={[Function]}
>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      Student Information
    </div>
    <div
      className="list-block inset sc-kAzzGY enMCSC"
    >
      <ul>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              account_circle
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="John Doe"
              type="text"
              value={undefined}
            />
          </div>
        </div>
        <div
          className="sc-ifAKCX item-content bqyzQp"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              email
            </i>
          </div>
          <div
            className="item-input sc-htpNat fYkctb"
          >
            <input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="john@school.edu"
              type="email"
              value=""
            />
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      School Information
    </div>
    <div
      className="list-block inset sc-kAzzGY enMCSC"
    >
      <ul>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a school..."
          >
            <select
              multiple={false}
              name="school"
              onBlur={undefined}
              onChange={[Function]}
              value="bruHy"
            >
              <option
                data-display-as="NJIT"
                value="bruHy"
              >
                New Jersey Institute of Technology, Newark
              </option>
              <option
                data-display-as="RU"
                value="5y7fh"
              >
                Rutgers University, New Brunswick
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  School
                </div>
                <div
                  className="item-after"
                />
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            className="item-link smart-select"
            data-searchbar="true"
            data-searchbar-placeholder="Search for a major..."
          >
            <select
              multiple={false}
              name="major"
              onBlur={undefined}
              onChange={[Function]}
              value="hvdMFjM"
            >
              <option
                data-display-as="CS"
                value="hvdMFjM"
              >
                Computer Science
              </option>
              <option
                data-display-as="IS"
                value="e2LejDg"
              >
                Information Systems
              </option>
            </select>
            <div
              className="item-content"
            >
              <div
                className="item-inner"
              >
                <div
                  className="item-title"
                >
                  Major
                </div>
                <div
                  className="item-after"
                />
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-content sc-jzJRlG dAvTTg"
    >
      <div
        className="card-content-inner"
      >
        <div
          className="header"
        >
          Attends Group Meetings
        </div>
        <div
          className="sc-fjdhpX cNVoic"
        >
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#D91518"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="1_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#D91518"
            >
              <label
                htmlFor="1_DRFJY9jd"
              >
                1
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#E6665B"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="2_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#E6665B"
            >
              <label
                htmlFor="2_DRFJY9jd"
              >
                2
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#DEB21C"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="3_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#DEB21C"
            >
              <label
                htmlFor="3_DRFJY9jd"
              >
                3
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#BCC75E"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="4_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#BCC75E"
            >
              <label
                htmlFor="4_DRFJY9jd"
              >
                4
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#A1BA37"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="5_DRFJY9jd"
              name="DRFJY9jd"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#A1BA37"
            >
              <label
                htmlFor="5_DRFJY9jd"
              >
                5
              </label>
            </div>
          </div>
        </div>
        <div
          className="footer"
        >
          <div>
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
             Awful
          </div>
          <div>
            Great 
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-content sc-jzJRlG dAvTTg"
    >
      <div
        className="card-content-inner"
      >
        <div
          className="header"
        >
          Contributes to Discussions
        </div>
        <div
          className="sc-fjdhpX cNVoic"
        >
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#D91518"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="1_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#D91518"
            >
              <label
                htmlFor="1_j84WAR77"
              >
                1
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#E6665B"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="2_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#E6665B"
            >
              <label
                htmlFor="2_j84WAR77"
              >
                2
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#DEB21C"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="3_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#DEB21C"
            >
              <label
                htmlFor="3_j84WAR77"
              >
                3
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#BCC75E"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="4_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#BCC75E"
            >
              <label
                htmlFor="4_j84WAR77"
              >
                4
              </label>
            </div>
          </div>
          <div
            checked={false}
            className="sc-jTzLTM fbgGmL"
            color="#A1BA37"
          >
            <input
              checked={false}
              className="sc-gqjmRU eYQGSB"
              id="5_j84WAR77"
              name="j84WAR77"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <div
              checked={false}
              className="sc-VigVT djmajI"
              color="#A1BA37"
            >
              <label
                htmlFor="5_j84WAR77"
              >
                5
              </label>
            </div>
          </div>
        </div>
        <div
          className="footer"
        >
          <div>
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
             Awful
          </div>
          <div>
            Great 
            <i
              className="material-icons icon"
            >
              trending_flat
            </i>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="content-block-title sc-bwzfXH cYrpQX"
    >
      Additional Information
    </div>
    <div
      className="list-block inset sc-kAzzGY kqlxdJ"
    >
      <ul>
        <li
          className="align-top"
        >
          <div
            className="sc-gZMcBi item-content irilpd"
          >
            <div
              className="item-media"
            >
              <i
                className="material-icons icon"
              >
                comment
              </i>
            </div>
            <div
              className="item-inner"
            >
              <div
                className="item-input"
              >
                <textarea
                  name="comment"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Your comment here..."
                />
              </div>
            </div>
          </div>
        </li>
        <div
          className="item-content"
        >
          <div
            className="item-media"
          >
            <i
              className="material-icons icon"
            >
              done
            </i>
          </div>
          <div
            className="item-inner"
          >
            <div
              className="sc-htoDjs item-title label kmibPL"
            >
              Recommend
            </div>
            <div
              className="sc-dnqmqq item-input kaKmBo"
            >
              <label
                className="label-switch"
              >
                <input
                  name="recommend"
                  onChange={[Function]}
                  type="checkbox"
                />
                <div
                  className="checkbox"
                />
              </label>
            </div>
          </div>
        </div>
      </ul>
    </div>
  </div>
  <div
    className="content-block  "
  >
    <button
      className="button button-big sc-bdVaJa dYzSUq"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
